### PR TITLE
Add an example for CupertinoPopupSurface

### LINKED ISSUE #150353: Add an example for CupertinoPopupSurface
### Use case

We have [CupertinoActionSheet](https://api.flutter.dev/flutter/cupertino/CupertinoActionSheet-class.html) with a form of action. Meanwhile, CupertinoPopupSurface is a customizable widget closer to [Sheets](https://developer.apple.com/design/human-interface-guidelines/sheets) under the Apple HIC. But there is no example for [CupertinoPopupSurface](https://api.flutter.dev/flutter/cupertino/CupertinoPopupSurface-class.html) currently. 

Providing a visual example of how to implement and utilize its properties will be helpful to users.

### Proposal

Add an example for CupertinoPopupSurface

### PR DESCRIPTION
**Pull request by huycozy**
### Demo

https://github.com/flutter/flutter/assets/104349824/61cd4c96-e01e-4fad-b270-acd7bb55d995

### Related issue

Fixes https://github.com/flutter/flutter/issues/150353

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/dialog.dart [hunk 1]
```diff
@@ -400,6 +400,14 @@ class _CupertinoAlertDialogState extends State<CupertinoAlertDialog> {
 /// Additionally, the white paint can be disabled to render a blurred rounded
 /// rectangle without any color (similar to iOS's volume control popup).
 ///
+/// {@tool dartpad}
+/// This sample shows how to use a [CupertinoPopupSurface].
+///	The [CupertinoPopupSurface] shows a model popup from the bottom of the screen.
```
Comment (bleroux) [inline packages/flutter/lib/src/cupertino/dialog.dart:405]:
```suggestion
/// The [CupertinoPopupSurface] shows a model popup from the bottom of the screen.
```
Consider removing extra indentation

**Review by bleroux — APPROVED** [review]
LGTM!

File: packages/flutter/lib/src/cupertino/dialog.dart [hunk 2]
```diff
@@ -400,6 +400,14 @@ class _CupertinoAlertDialogState extends State<CupertinoAlertDialog> {
 /// Additionally, the white paint can be disabled to render a blurred rounded
 /// rectangle without any color (similar to iOS's volume control popup).
 ///
+/// {@tool dartpad}
+/// This sample shows how to use a [CupertinoPopupSurface].
+/// The [CupertinoPopupSurface] shows a model popup from the bottom of the screen.
```
Comment (TahaTesser) [inline packages/flutter/lib/src/cupertino/dialog.dart:405]:
```suggestion
/// This sample shows how to use a [CupertinoPopupSurface]. The [CupertinoPopupSurface]
/// shows a model popup from the bottom of the screen.
```
This line break looks unusual here with short sentence.

**Review by TahaTesser — APPROVED** [review]
LGTM

**Pull request merged**